src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -110,46 +110,43 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 	private processWorkspaceRecommendations(extensionsContent: IExtensionsContent): TPromise<string[]> {
 		const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
 
-		if (extensionsContent && extensionsContent.recommendations) {
+		if (extensionsContent && extensionsContent.recommendations && extensionsContent.recommendations.length) {
+			let countBadRecommendations = 0;
+			let badRecommendationsString = '';
 			let filteredRecommendations = extensionsContent.recommendations.filter((element, position) => {
-				return extensionsContent.recommendations.indexOf(element) === position && regEx.test(element);
-			});
-
-			if (filteredRecommendations.length !== extensionsContent.recommendations.length) {
-				let badlyFormattedRecommendations = extensionsContent.recommendations.filter(element => {
-					return filteredRecommendations.indexOf(element) === -1;
-				});
-
-				let badRecommendationsString = '';
-				badlyFormattedRecommendations.forEach(element => {
-					badRecommendationsString += element + '\n';
-				});
+				if (!regEx.test(element)) {
+					countBadRecommendations++;
+					badRecommendationsString += `${element} (bad format) Expected: <provider>.<name>\n`;
+					return false;
+				} else if (extensionsContent.recommendations.indexOf(element) !== position) {
+					countBadRecommendations++;
+					badRecommendationsString += `${element} (duplicate)\n`;
+					return false;
+				}
 
-				console.log(badlyFormattedRecommendations.length +
-					' extension(s) in workspace recommendations are incorrectly formatted. Expected <publisher>.<extension name>:\n' +
-					badRecommendationsString);
-			}
+				return true;
+			});
 
 			return this._galleryService.query({ names: filteredRecommendations }).then(pager => {
 				let page = pager.firstPage;
 				let validRecommendations = page.map(extension => {
 					return extension.identifier.id.toLowerCase();
 				});
 
-
 				if (validRecommendations.length !== filteredRecommendations.length) {
-					let unknownExtensions = filteredRecommendations.filter(element => {
-						return validRecommendations.indexOf(element.toLowerCase()) === -1;
-					});
-
-					let unknownExtensionsString = '';
-					unknownExtensions.forEach(element => {
-						unknownExtensionsString += element + '\n';
+					filteredRecommendations.forEach(element => {
+						if (validRecommendations.indexOf(element.toLowerCase()) === -1) {
+							countBadRecommendations++;
+							badRecommendationsString += `${element} (not found in gallery)\n`;
```

Comment:
Use the word `marketplace` instead of `gallery`. I know, its called the gallery service, but the website users can go search for extension is called `marketplace` (powered by the same `gallery service` :) )